### ISSUE TITLE
The index.js within themes is ignored during the build process

### ISSUE THREAD

**Issue by SebastianNiemann** [vuejs/vuepress#942]
#### Version

1.0.0-alpha.12

#### Steps to reproduce

Add the following `index.js` to a theme:

```
process.exit(1);

```

#### What is expected?

The built process should fail with exit code 1, proving that `index.js` was called.

#### What is actually happening?

The built process is successful, fully ignoring the `index.js` and therefore any options within it (esp. `plugins`).

#### Other relevant information

Compared to #928, the theme is included by simply stating its file path, instead of a dependency package

**Comment by ulivz** [comment]
Cannot reproduce it, could you provide more info?

**Comment by SebastianNiemann** [comment]
I just created a reproduction repo to demonstrate it: https://github.com/SebastianNiemann/vuepress-issue-942

Within this, I am using a local theme. Not sure what you meant by *laptop's version*, but if it is about the OS, I observed this under Windows 10.

**Comment by ulivz** [comment]
Maybe you should read the docs carefully, you just used a way that we didn't mention in the document.

Move the theme directory to `.vuepress` directory instead.

See: [Directory Structure](http://localhost:8080/guide/directory-structure.html).

**Comment by SebastianNiemann** [comment]
Understood.

Just to give a little bit reasoning how we got here into the first place:

We actually use absolute paths as theme directories within our projects (especially since everything works fine, despite this `index.js` issue), in cases where multiple sites use the same theme and everything is placed in the same project, avoiding `kill process && yarn upgrade && yarn serve` to reload changes in the theme during development.

However, it seemed that our understanding about the `theme` config option (https://vuepress.vuejs.org/config/#theme) came from an unclear understanding whether paths would be acceptable, especially since they worked flawless before.

To make sure, we looked into the source (currently at `@vuepress/core/lib/prepare/loadTheme.js`) where the source documentation of `loadTheme` states:
```
 * Resolve theme.
 *
 *   Resolving Priority:
 *
 *   1. If the theme was a absolute path and that path exists, respect it
 *      as the theme directory.
 *   2. If 'theme' directory located at vuepressDir exists, respect it as
 *      the theme directory.
 *   3. If 'theme' was a shortcut string, resolve it from deps.
```

Given that the top priority was to resolve  absolute paths as theme paths (and all the code within `loadTheme` to support this), we where unfortunately under the impression that paths where (already?) supported. However, this was just meant to share our experience, no biggie.

---

@ulivz Having all that said, I would suggest a more clear config documentation (especially since try&error would suggest that paths are supported), or happily volunteer to fully support absolute paths as theme directories 😃
